### DISCUSSION TITLE
Signals specific to a workspace

### DISCUSSION THREAD

**Discussion by ktriple** [datalust/seq-tickets#1995] [Help]
We'd like to create shared signals for specific workspaces.

Currently when you create a shared signal, every workspace sees that signal. That clutters all the workspaces with signals. We have a Workspace filter on Users and API keys, to keep things tidy / secure.

It is possible to manually remove a signal from a workspace, but you'd have to do that for every workspace and each signal that gets added. That's not a workflow we can work with.

Are we overlooking something, or is it possible to create an extension for this (I'd be happy to look into UI extension)?

**Comment by nblumhardt** [comment]
Hi @ktriple, thanks for dropping us a line.

When a new shared signal is created, it shouldn't appear in any workspaces apart from the one that the creating user has active. Are you seeing different behavior than that?

Users will still be able to see all shared signals when they click in/search the signal bar, but that doesn't sound like the problem given your manual workaround.

I might not be understanding you correctly; would it be possible to follow up with some detailed steps/screenshots of the problematic behavior?

Many thanks,
Nick

**Comment by ktriple** [comment]
@nblumhardt 

Thanks for getting back with me, I think I understand the behavior better now. All signals are always available under the "+ X More" button, and that's what confused us.

![Screenshot 2023-10-23 at 10 27 00](https://github.com/datalust/seq-tickets/assets/44257767/41c6d9b5-f8d8-4707-b963-1384ed1340d3)

Personally I'd say it might be a good idea to separate signals more, as now you see "+X More" all the time in your workspace, suggesting you miss signals in that same workspace that you should click open. And if you do, than you get all signals that exist (your own / shared).

You'd want to easily see that something is a workspace signal, perhaps just with a separator under them or headers "Workspace" / "Other" or place the "+ X More" somewhere at the bottom.

**Reply by nblumhardt** [reply]
Thanks for the feedback @ktriple 👍
